src/main/groovy/edu/oregonstate/mist/locations/db/ArcGisDAO.groovy

```diff
@@ -60,12 +60,17 @@ class ArcGisDAO {
                 "attributes"
         )
 
-        mergeArcGisData(centroidData, genderInclusiveRRData)
+        mergeArcGisData(centroidData, genderInclusiveRRData, buildings)
     }
 
     private HashMap<String, ArcGisLocation> mergeArcGisData(
             HashMap<String, ArcGisLocation> centroidData,
-            HashMap<String, ArcGisLocation> genderInclusiveRRData) {
+            HashMap<String, ArcGisLocation> genderInclusiveRRData, List<FacilLocation> buildings) {
+
+        def buildingIDs = buildings*.bldgID
+        def centroidDataToBeRemoved =  centroidData.collect { it.value.bldID } - buildingIDs
+        logger.warn("Buildings found in centroid, but not in AIM: " + centroidDataToBeRemoved)
+        centroidData.entrySet().removeIf { ! buildingIDs.contains(it.value.bldID) }
 
         genderInclusiveRRData.each {id, value ->
             if (centroidData[id]) {
```

Comment:
Same goes for geometries data. I found some buildings that we have geometries for, but the geometries are not being merged because they're not in the centroid data (but are in the facil location data).